src/feature/control/control_getinfo.c

```diff
@@ -325,47 +325,32 @@ getinfo_helper_current_time(control_connection_t *control_conn,
   return 0;
 }
 
-/**
- * Switch between microdesc vs networkstatus descriptor dumps
- * Assumes that question would be either:
- * - dir/status-vote/current/consensus-microdesc
- * or
- * - dir/status-vote/current/consensus
- */
+/** Fetch network-status or micro-descriptor
+ * returns: 0 on success -1 on error. */
 STATIC int
-getinfo_helper_current_consensus(const char* question,
-                          char** answer,
-                          const char** errmsg)
+getinfo_helper_current_consensus(int flavor,
```

Comment:
No need to pass both `flavor` and `flavor_name`. That's redundant.

Please pass just one, and derive the other in the function.